### PR TITLE
executor,sessionctx: enable coprocessor paging and make TestCoprocessorPagingSize stable

### LINKED ISSUE #38158: unstable test TestCoprocessorPagingSize
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
    distsql_test.go:467: 
        	Error Trace:	/home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/sandbox/processwrapper-sandbox/6362/execroot/__main__/bazel-out/k8-fastbuild/bin/executor/executor_test_/executor_test.runfiles/__main__/executor/distsql_test.go:467
        	Error:      	"2" is not greater than "2"
        	Test:       	TestCoprocessorPagingSize
```

https://do.pingcap.net/jenkins/blue/rest/organizations/jenkins/pipelines/pingcap/pipelines/tidb/pipelines/ghpr_unit_test/runs/1456/nodes/49/steps/52/log/?start=0

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #38158

Problem Summary:

`TestCoprocessorPagingSize` assume the coprocessor paging protocol, otherwise it become unstable.

In https://github.com/pingcap/tidb/pull/37920, the `tidb_enable_paging` default value is changed to `false`, to workaround a TPCH Q17 performance regression https://github.com/pingcap/tidb/issues/37768.

Now that I've found the root cause https://github.com/pingcap/tidb/pull/38012 of TPCH Q17, it can be change back.

### What is changed and how it works?

Set `@@tidb_enable_paging` default to true and make `TestCoprocessorPagingSize` code more stable.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
@tiancaiamao PTAL
```
    result.go:49: 
        	Error Trace:	/home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/sandbox/processwrapper-sandbox/6358/execroot/__main__/bazel-out/k8-fastbuild/bin/planner/core/core_test_/core_test.runfiles/__main__/planner/core/result.go:49
        	            				/home/jenkins/.tidb/tmp/63a9840cd0739f2c243bb46478607469/sandbox/processwrapper-sandbox/6358/execroot/__main__/bazel-out/k8-fastbuild/bin/planner/core/core_test_/core_test.runfiles/__main__/planner/core/physical_plan_test.go:2341
        	Error:      	Not equal: 
        	            	expected: "[Projection 250.00 root  test.ta.name, Column#13]\n[└─Apply 250.00 root  CARTESIAN left outer join]\n[  ├─IndexLookUp(Build) 250.00 root  ]\n[  │ ├─IndexRangeScan(Build) 250.00 cop[tikv] table:ta, index:idx_ta_name(name) range:[\"chad999\",\"chad99:\"), keep order:false, stats:pseudo]\n[  │ └─TableRowIDScan(Probe) 250.00 cop[tikv] table:ta keep order:false, stats:pseudo]\n[  └─MaxOneRow(Probe) 1.00 root  ]\n[    └─HashAgg 1.00 root  funcs:sum(Column#17)->Column#13]\n[      └─IndexLookUp 1.00 root  ]\n[        ├─IndexRangeScan(Build) 10.00 cop[tikv] table:tb, index:idx_tb_id(id) range: decided by [eq(test.ta.id, test.tb.id)], keep order:false, stats:pseudo]\n[        └─HashAgg(Probe) 1.00 cop[tikv]  funcs:sum(test.tb.code)->Column#17]\n[          └─TableRowIDScan 10.00 cop[tikv] table:tb keep order:false, stats:pseudo]\n"
        	            	actual  : "[Projection 250.00 root  test.ta.name, Column#13]\n[└─Apply 250.00 root  CARTESIAN left outer join]\n[  ├─IndexLookUp(Build) 250.00 root  ]\n[  │ ├─IndexRangeScan(Build) 250.00 cop[tikv] table:ta, index:idx_ta_name(name) range:[\"chad999\",\"chad99:\"), keep order:false, stats:pseudo]\n[  │ └─TableRowIDScan(Probe) 250.00 cop[tikv] table:ta keep order:false, stats:pseudo]\n[  └─MaxOneRow(Probe) 1.00 root  ]\n[    └─StreamAgg 1.00 root  funcs:sum(Column#21)->Column#13]\n[      └─Projection 10.00 root  cast(test.tb.code, decimal(10,0) BINARY)->Column#21]\n[        └─IndexLookUp 10.00 root  ]\n[          ├─IndexRangeScan(Build) 10.00 cop[tikv] table:tb, index:idx_tb_id(id) range: decided by [eq(test.ta.id, test.tb.id)], keep order:false, stats:pseudo]\n[          └─TableRowIDScan(Probe) 10.00 cop[tikv] table:tb keep order:false, stats:pseudo]\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,7 +6,7 @@
        	            	 [  └─MaxOneRow(Probe) 1.00 root  ]
        	            	-[    └─HashAgg 1.00 root  funcs:sum(Column#17)->Column#13]
        	            	-[      └─IndexLookUp 1.00 root  ]
        	            	-[        ├─IndexRangeScan(Build) 10.00 cop[tikv] table:tb, index:idx_tb_id(id) range: decided by [eq(test.ta.id, test.tb.id)], keep order:false, stats:pseudo]
        	            	-[        └─HashAgg(Probe) 1.00 cop[tikv]  funcs:sum(test.tb.code)->Column#17]
        	            	-[          └─TableRowIDScan 10.00 cop[tikv] table:tb keep order:false, stats:pseudo]
        	            	+[    └─StreamAgg 1.00 root  funcs:sum(Column#21)->Column#13]
        	            	+[      └─Projection 10.00 root  cast(test.tb.code, decimal(10,0) BINARY)->Column#21]
        	            	+[        └─IndexLookUp 10.00 root  ]
        	            	+[          ├─IndexRangeScan(Build) 10.00 cop[tikv] table:tb, index:idx_tb_id(id) range: decided by [eq(test.ta.id, test.tb.id)], keep order:false, stats:pseudo]
        	            	+[          └─TableRowIDScan(Probe) 10.00 cop[tikv] table:tb keep order:false, stats:pseudo]
        	            	 
        	Test:       	TestNoDecorrelateHint
        	Messages:   	sql:explain format = 'brief' SELECT ta.NAME,(SELECT /*+ no_decorrelate() */ sum(tb.CODE) FROM tb WHERE ta.id = tb.id) tb_sum_code FROM ta WHERE ta.NAME LIKE 
```

File: planner/core/testdata/plan_suite_out.json [hunk 1]
```diff
@@ -6026,11 +6026,11 @@
           "  │ ├─IndexRangeScan(Build) 250.00 cop[tikv] table:ta, index:idx_ta_name(name) range:[\"chad999\",\"chad99:\"), keep order:false, stats:pseudo",
           "  │ └─TableRowIDScan(Probe) 250.00 cop[tikv] table:ta keep order:false, stats:pseudo",
           "  └─MaxOneRow(Probe) 1.00 root  ",
-          "    └─HashAgg 1.00 root  funcs:sum(Column#17)->Column#13",
-          "      └─IndexLookUp 1.00 root  ",
-          "        ├─IndexRangeScan(Build) 10.00 cop[tikv] table:tb, index:idx_tb_id(id) range: decided by [eq(test.ta.id, test.tb.id)], keep order:false, stats:pseudo",
-          "        └─HashAgg(Probe) 1.00 cop[tikv]  funcs:sum(test.tb.code)->Column#17",
-          "          └─TableRowIDScan 10.00 cop[tikv] table:tb keep order:false, stats:pseudo"
+          "    └─StreamAgg 1.00 root  funcs:sum(Column#21)->Column#13",
```
Comment (XuHuaiyu) [inline planner/core/testdata/plan_suite_out.json:6029]: Is this affected by paging?

Comment (tiancaiamao) [reply planner/core/testdata/plan_suite_out.json:6029]: Yes, it seems that the optimizer consider the new plan is better with paging available.

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**